sdk/spring/spring-cloud-azure-feature-management/README.md

```diff
@@ -192,10 +192,136 @@ feature-management:
+              pattern:
+                type: "Daily",
+                interval: 1
+              range:
+                type: "NoEnd"
+```
+
+The `Recurrence` settings is made up of two parts: `Pattern` (how often the time window will repeat) and `Range` (for how long the recurrence pattern will repeat).
+#### Recurrence Pattern
+
+There are two possible recurrence pattern types: `Daily` and `Weekly`. For example, a time window could repeat "every day", "every 3 days", "every Monday" or "on Friday per 2 weeks".
+
+Depending on the type, certain fields of the `Pattern` are required, optional, or ignored.
+
+- `Daily`
+
+  The daily recurrence pattern causes the time window to repeat based on a number of days between each occurrence.
+
+  | Property | Relevance | Description |
+      |----------|-----------|-------------|
+  | **Type** | Required | Must be set to `Daily`. |
+  | **Interval** | Optional | Specifies the number of days between each occurrence. Default value is 1. |
+
+- `Weekly`
+
+  The weekly recurrence pattern causes the time window to repeat on the same day or days of the week, based on the number of weeks between each set of occurrences.
+
+  | Property | Relevance | Description |
+      |----------|-----------|-------------|
+  | **Type** | Required | Must be set to `Weekly`. |
+  | **DaysOfWeek** | Required | Specifies on which day(s) of the week the event occurs. |
+  | **Interval** | Optional | Specifies the number of weeks between each set of occurrences. Default value is 1. |
+  | **FirstDayOfWeek** | Optional | Specifies which day is considered the first day of the week. Default value is `Sunday`. |
+
+  The following example will repeat the time window every other Monday and Tuesday
+
+    ```yaml
+    pattern:
+        type: "Weekly",
+        interval: 1,
+        daysOfWeek: 
+          - Monday
+          - TuesDay
+    ```
+  **Note:** `Start` must be a valid first occurrence which fits the recurrence pattern. Additionally, the duration of the time window cannot be longer than how frequently it occurs. For example, it is invalid to have a 25-hour time window recur every day.
+
+#### Recurrence Range
+
+There are three possible recurrence range type: `NoEnd`, `EndDate` and `Numbered`.
+
+- `NoEnd`
+
+  The `NoEnd` range causes the recurrence to occur indefinitely.
+
+  | Property | Relevance | Description |
+      |----------|-----------|-------------|
+  | **Type** | Required | Must be set to `NoEnd`. |
+
+- `EndDate`
+
+  The `EndDate` range causes the time window to occur on all days that fit the applicable pattern until the end date.
+
+  | Property | Relevance | Description |
+      |----------|-----------|-------------|
+  | **Type** | Required | Must be set to `EndDate`. |
+  | **EndDate** | Required | 	Specifies the date time to stop applying the pattern. Note that as long as the start time of the last occurrence falls before the end date, the end time of that occurrence is allowed to extend beyond it. |
```

Comment:
Sorry, I didn't get. 

> What if I want EndDate to be a hard cutoff

What do you mean by this? The `EndDate` should be required when you use `EndDate` recurrence range type.

Reply:
Sorry, what I meant by optional is the "Note" part. I might want a time window to have a hard end at the `endDate`, not a "I will not repeat after" option like we have here.

Reply:
Hi Matthew, I could get your point. And it also reasonable that we should have a hard cut off. But if we need to change this logic, we may need to involve more people for discussion since dotnet is already merged and close to release. 
Besides, we choose the current solution because it's the way that outlook does. Not sure if you have tried to create a repeated meeting in outlook. In outlook, it only promise that the start time of the meeting is before the end date, not promise the end time of the meeting is after the start date.